homework_02/victor_dimitrov_04/views.py

```diff
@@ -0,0 +1,43 @@
+from django.shortcuts import render
+import pytz
+import datetime
+
+# Create your views here.
+
+import requests
+
+def trains(request):
+
+    initial_filters = {
+        'page[offset]': '0',
+        'page[limit]': '50',
+        'include': 'trip,schedule',
+        'filter[stop]': 'place-north',
+        'filter[route_type]': '2',
+    }
+
+    response = requests.get('https://api-v3.mbta.com/predictions', params=initial_filters).json()
+
+    trains = []
+
+    for element in response['data']:
+        train_info = {}
+        train_info['status'] = element['attributes']['status']
+        train_info['route'] = element['relationships']['route']['data']['id'][3:]
+
+        time_filters = {
+            'include': 'prediction',
+            'filter[route]': element['relationships']['route']['data']['id'],
+            'filter[direction_id]' : 0
+        }
+
+        time_response = requests.get('https://api-v3.mbta.com/schedules', params=time_filters).json()
```

Comment:
The "Schedules" object can be placed in the "includes" parameter instead of sending so many requests.